crates/sc-consensus-subspace-rpc/src/lib.rs

```diff
@@ -798,4 +835,66 @@ where
 
         Ok(last_segment_headers)
     }
+
+    fn subscribe_archived_object_mappings(&self, pending: PendingSubscriptionSink) {
+        let cached_archived_segment = Arc::clone(&self.cached_archived_segment);
+
+        // Ignore acknowledgment sender: it is already handled by subscribe_archived_segment_header()
+        let stream = self
+            .archived_segment_notification_stream
+            .subscribe()
+            .flat_map(move |archived_segment_notification| {
+                let archived_segment = archived_segment_notification.archived_segment;
+
+                cached_archived_segment
+                    .lock()
+                    .replace(CachedArchivedSegment::Weak(Arc::downgrade(
+                        &archived_segment,
+                    )));
+
+                let object_mappings = archived_segment.object_mappings();
+
+                stream::iter(object_mappings)
+            });
+
+        let fut = async move {
+            pipe_from_stream(pending, stream).await;
+        };
+
+        self.subscription_executor.spawn(
+            "subspace-archived-object-mappings-subscription",
+            Some("rpc"),
+            fut.boxed(),
+        );
+    }
+
+    // TODO:
```

Comment:
As explained above this will not really work and I'm not 100% use why it is needed. On-demand derivation is a useful scenario, but I'd probably tackle it separately.

Reply:
I think I’ll just delete this code for now, and come back to it later if needed.

That also means we don’t need to do the refactor.